packages/react-server/src/ReactFizzHooks.js

```diff
@@ -398,13 +398,14 @@ export function useReducer<S, I, A>(
         } while (update !== null);
 
         // $FlowFixMe[incompatible-use] found when upgrading Flow
-        workInProgressHook.memoizedState = newState;
-
-        return [newState, dispatch];
+        workInProgressHook.memoizedState = [newState, dispatch];
```

Comment:
I thought about `.memoizedState[0] = newState`, but that would make the return value of `useState` stable across renders, even when there are changes, and that would be conceptually wrong for react.

Tbh my true aim was to have a state hook with a stable value (and absolutely no memory allocations beyond initialization), but it would have an API like this:

```javascript
const enabled = useStableState(false); // { get: () => boolean, set: (v: boolean) => void }
return <button onClick=(() => enabled.set(!enabled.get())>{enabled.get() ? 'enabled' : 'disabled'}</button>
```

However, I fear it won't be possible without adding it directly in React as I need access to the internals to implement it properly.